docs/blog/2018-10-16-unstructured-data/index.md

```diff
@@ -0,0 +1,203 @@
+
+  // Create a page that lists all Pokémon.
+  createPage({
+    path: `/`,
+    component: require.resolve("./src/templates/all-pokemon.js"),
+    context: { allPokemon },
+  })
+
+  // Create a page for each Pokémon.
+  allPokemon.forEach(pokemon => {
+    createPage({
+      path: `/pokemon/${pokemon.name}/`,
+      component: require.resolve("./src/templates/pokemon.js"),
+      context: { pokemon },
+    })
+  })
+}
+```
+
+#### 6. Create a page [for each ability of each Pokémon](https://github.com/jlengstorf/gatsby-with-unstructured-data/blob/0a91d87b9d4d24a0e6b04b33cc271e054b7467b6/gatsby-node.js#L40).
+
+```javascript{19-26}:title=gatsby-node.js
+exports.createPages = async ({ actions: { createPage } }) => {
+  const allPokemon = await getPokemonData(["pikachu", "charizard", "squirtle"])
+
+  // Create a page that lists all Pokémon.
+  createPage({
+    path: `/`,
+    component: require.resolve("./src/templates/all-pokemon.js"),
+    context: { allPokemon },
+  })
+
+  // Create a page for each Pokémon.
+  allPokemon.forEach(pokemon => {
+    createPage({
+      path: `/pokemon/${pokemon.name}/`,
+      component: require.resolve("./src/templates/pokemon.js"),
+      context: { pokemon },
+    })
+
+    // Create a page for each ability of the current Pokémon.
+    pokemon.abilities.forEach(ability => {
+      createPage({
+        path: `/pokemon/${pokemon.name}/ability/${ability.name}/`,
+        component: require.resolve("./src/templates/ability.js"),
+        context: { pokemon, ability },
+      })
+    })
+  })
+}
+```
+
+For each type of page, we are invoking the `createPage` action, and supplying it with our desired path, React component, and data (as props on the context).
+
+## When might using "unstructured data" make sense?
+
+You may find this approach useful when you have a use case where using Gatsby's data layer feels like swatting flies with a sledgehammer -- it'll get the job done, but perhaps feels too heavy-handed for your project scope.
+
+## The pros of using unstructured data
+
+- The approach is familiar and comfortable, especially if you’re new to GraphQL
+- There’s no intermediate step: you fetch some data, then build pages with it
+
+## The tradeoffs of foregoing Gatsby's data layer
+
+Using Gatsby's data layer provides the following benefits:
+
+- Enables you to declaratively specify what data a page component needs
+- Eliminates frontend data boilerplate — no need to worry about requesting & waiting for data. Just ask for the data you need with a GraphQL query and it’ll show up when you need it
+- Pushes frontend complexity into queries — many data transformations can be done at build-time within your GraphQL queries
+- It’s the perfect data querying language for the often complex/nested data dependencies of modern applications
+- Improves performance by removing data bloat — GraphQL is a big part of why Gatsby is so fast as it enables lazy-loading the exact data in the exact form each view needs
```

Comment:
More tradeoffs (just need nicer way to put this):
If you want to change data passed to pages you need to restart dev server (consider adding "see other pokemons" section to pokemon detail view - you would need to pass `allPokemons` to subpages and restart dev server - with queries you just add query for all pokemons and it will hot reload)